ocean/src/social.rs

```
@@ -17,7 +22,28 @@ impl SocialNetwork {
     * Result enum
     */
     pub fn add_friend(&mut self, crab_id: String, friend_id: String) -> Result<(), String> {
-        unimplemented!();
+        if crab_id == friend_id {
+            // Prevent a crab from being friends with itself
+            return Err("A crab cannot befriend itself".to_string());
+        }
+
+        match self.friendships.get_mut(&crab_id) {
+            Some(friends) => {
+                if friends.contains(&friend_id) {
+                    // The friend is already in the list
+                    Err("Already friends".to_string())
+                } else {
```

We could return the Err directly and avoid the else { } syntax